app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -0,0 +1,21 @@
+module HasMonitoringManagerMixin
```

Comment:
Thanks for the heads up @durandom. I experience some weird stuff (that was solved [passing parent_manager to build](https://github.com/ManageIQ/manageiq/pull/15354/files#diff-4f413339a0e0be4946fb726fb8dacb9aR9)). We should be good with the first issue and the second one (we use on_create, and Monitoring manager always has a parent_manager).

I should probably change ensure_manager to ensure_X_manager.

I would appreciate it if you can take a look on the manageiq-providers-kubernetes [PR](https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/38) adding the concrete manager.